pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -143,12 +143,27 @@ cdef inline _Timestamp create_timestamp_from_ts(
     """ convenience routine to construct a Timestamp from its parts """
     cdef:
         _Timestamp ts_base
-
-    ts_base = _Timestamp.__new__(Timestamp, dts.year, dts.month,
+        int64_t pass_year = dts.year
+
+    # We pass year=1970/1972 here and set year below because with non-nanosecond
+    #  resolution we may have datetimes outside of the stdlib pydatetime
+    #  implementation bounds, which would raise.
+    # NB: this means the C-API macro PyDateTime_GET_YEAR is unreliable.
+    if 1 <= pass_year <= 9999:
```

Comment:
Worth being specific with `datetime.min.year <= pass_year <= datetime.max.year`?

Reply:
i think the comment is enough, dont want to add more lookup overhead